packages/service-integration/tests/test__usecase_jupytermath.py

```diff
@@ -0,0 +1,110 @@
+
+
+def _git_version() -> str:
+    return subprocess.run(
+        ["git", "--version"],
+        capture_output=True,
+        encoding="utf8",
+        check=True,
+    ).stdout.strip()
+
+
+def _git_shallow_clone(repository: str, directory: Path):
+    subprocess.run(
+        [
+            "git",
+            "clone",
+            "--depth=1",
+            "--verbose",
+            "--progress",
+            repository,
+            f"{directory}",
+        ],
+        check=True,
+    )
+    return directory
+
+
+@pytest.fixture(scope="module")
+def jupytermath_repo(tmp_path_factory: TempPathFactory) -> Path:
+    print("Running git", _git_version())
+    tmp_path = tmp_path_factory.mktemp("jupytermath_repo")
+    repo_dir = _git_shallow_clone(
+        "https://github.com/ITISFoundation/jupyter-math", tmp_path / "jupyter-math"
+    )
+    assert repo_dir.exists()
+    os.system(f"ls -la {repo_dir}")
+    return repo_dir
+
+
+@pytest.fixture
+def run_program_in_repo(tmp_path: Path, jupytermath_repo: Path) -> Iterable[Callable]:
+    runner = CliRunner()
+    with runner.isolated_filesystem(temp_dir=tmp_path) as workdir:
+        shutil.copytree(jupytermath_repo, workdir, dirs_exist_ok=True)
+        os.system(f"ls -la {workdir}")
+
+        def _invoke(*cmd) -> tuple[Path, Result]:
+            print(
+                "RUNNING",
+                "osparc-service-integrator",
+                " ".join(cmd),
+                f"at {workdir=}",
+            )
+            print("ENV:", runner.make_env())
+            #
+            result = runner.invoke(cli.app, list(cmd))
+            return Path(workdir), result
+
+        yield _invoke
+
+
+def test_ooil_compose_wo_arguments(
+    run_program_in_repo: Callable[..., tuple[Path, Result]],
+):
+    # After https://github.com/ITISFoundation/osparc-simcore/pull/3433#pullrequestreview-1138481844
+    #
+    # this test reproduces the calls with ooil in jupyter-math repo:
+    #
+    # SEE https://github.com/ITISFoundation/jupyter-math/blob/ad51f531548e88afa3ffe3b702a89d159ad8be7f/Makefile#L29
+    # SEE https://github.com/ITISFoundation/jupyter-math/blob/ad51f531548e88afa3ffe3b702a89d159ad8be7f/.github/workflows/check-image.yml#L18
+    #
+
+    # NOTE: defaults searches for configs in .osparc/**/metadata.yml
+    workdir, result = run_program_in_repo(
+        "compose",
+    )
+    assert result.exit_code == os.EX_OK, result.output
+
+    # should produce compose-specs file
+    compose_spec_path = workdir / "docker-compose.yml"
+    assert compose_spec_path.exists()
+
+    # some checks compose-specs content
+    compose_spec = yaml.safe_load(compose_spec_path.read_text())
+    print(json.dumps(compose_spec, indent=1))
+
+    assert "jupyter-math" in compose_spec["services"]
+
+    labels = compose_spec["services"]["jupyter-math"]["build"]["labels"]
+    assert (
+        labels["org.label-schema.vcs-url"]
+        == "https://github.com/ITISFoundation/jupyter-math"
+    )
```

Comment:
You could add an additional layer of testing to make sure there are not regressions.

Run [make compose-spec](https://github.com/ITISFoundation/jupyter-math/blob/ad51f531548e88afa3ffe3b702a89d159ad8be7f/Makefile#L25) in the cloned repository. Then compare the generated `docker-compose.yaml` file with the locally generated one.

Reply:
OK, i get your point but I have done some modifications to your proposal:

Pulling ``Unable to find image 'itisfoundation/ci-service-integration-library:v1.0.1-dev-25' locally`` takes a lot of time for this type of tests. I have instead pinned the repo commit and taken the resulting ``docker-compose``and put it in ``tests/data/docker-compose_jupyter-math_v1-0-1-dev-25.yml``

In addition, take into account that this test is to ensure that ``ooil`` HEAD works as before this date!  Nonetheless, newer versions of ``ooil`` might produce in time different ``docker-compose`` outputs to those generated by an older version. This means that the comparison is far more subtle than just diff-ing two docker-compose ! We have to check instead that those are "equivalent" (since, as I mentioned, they well be produced by different versions of ``ooil``)

IMO, the reasons above suggest that the test, as you suggested, is more suited on the client side instead of here.

Here I will add a variation of your suggestion that I think reasonably cover our purpose.

Reply:
Perfect, this will do the job as well. Thanks for taking care of it!
This should give us some guarantees about the process!